src/app/network/crypto.ts

```diff
@@ -32,16 +32,17 @@ export function generateHMAC(
   return hmac.digest();
 }
 
-function getDeterministicKey(key: string, data: string): Buffer {
+async function getDeterministicKey(key: string, data: string): Promise<string> {
```

Comment:
Remove async and await, if you return a Promise, that's enough

Reply:
Done